packages/react-native-session-replay/android/src/test/kotlin/com/datadog/reactnative/sessionreplay/ReactTextPropertiesResolverTest.kt

```diff
@@ -0,0 +1,230 @@
+package com.datadog.reactnative.sessionreplay
+
+import android.widget.TextView
+import com.datadog.android.sessionreplay.model.MobileSegment
+import com.datadog.reactnative.sessionreplay.ReactTextPropertiesResolver.Companion.COLOR_FIELD_NAME
+import com.datadog.reactnative.sessionreplay.ReactTextPropertiesResolver.Companion.FONT_FAMILY_FIELD_NAME
+import com.datadog.reactnative.sessionreplay.ReactTextPropertiesResolver.Companion.TEXT_ATTRIBUTES_FIELD_NAME
+import com.datadog.reactnative.sessionreplay.utils.ReflectionUtils
+import com.datadog.reactnative.sessionreplay.utils.formatAsRgba
+import com.datadog.tools.unit.GenericAssert.Companion.assertThat
+import com.facebook.react.bridge.ReactContext
+import com.facebook.react.uimanager.Spacing
+import com.facebook.react.uimanager.UIManagerModule
+import com.facebook.react.views.text.TextAttributes
+import com.facebook.react.views.view.ReactViewBackgroundDrawable
+import fr.xgouchet.elmyr.Forge
+import fr.xgouchet.elmyr.annotation.IntForgery
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.kotlin.whenever
+import org.mockito.quality.Strictness
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+internal class ReactTextPropertiesResolverTest {
+    private lateinit var testedResolver: ReactTextPropertiesResolver
+
+    @Mock
+    lateinit var mockReactContext: ReactContext
+
+    @Mock
+    lateinit var mockUiManagerModule: UIManagerModule
+
+    @Mock
+    lateinit var mockReflectionUtils: ReflectionUtils
+
+    @Mock
+    lateinit var mockWireframe: MobileSegment.Wireframe.TextWireframe
+
+    @Mock
+    lateinit var mockTextView: TextView
+
+    @Mock
+    lateinit var mockReactViewBackgroundDrawable: ReactViewBackgroundDrawable
+
+    private lateinit var fakeWireframe: MobileSegment.Wireframe.TextWireframe
+
+    @BeforeEach
+    fun `set up`(forge: Forge) {
+        fakeWireframe = MobileSegment.Wireframe.TextWireframe(
+            id = forge.aPositiveLong(),
+            x = forge.aPositiveLong(),
+            y = forge.aPositiveLong(),
+            width = forge.aPositiveLong(),
+            height = forge.aPositiveLong(),
+            clip = null,
+            shapeStyle = null,
+            border = null,
+            text = forge.anAsciiString(),
+            MobileSegment.TextStyle(
+                family = "Serif",
+                size = forge.aPositiveLong(),
+                "#000000"
+            ),
```

Comment:
let's extract this to the `ForgeryFactory`, maybe?

Reply:
I'll extract this, but I'm a bit concerned that it duplicates logic in the native sdk. I'm wondering if we shouldn't change the visibility of the forgery factories there

Reply:
you can check how the code in `testFixtures` is shared across the modules